core/src/main/java/ai/timefold/solver/core/impl/move/streams/dataset/DatasetSessionFactory.java

```diff
@@ -0,0 +1,64 @@
+package ai.timefold.solver.core.impl.move.streams.dataset;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Consumer;
+
+import ai.timefold.solver.core.impl.bavet.NodeNetwork;
+import ai.timefold.solver.core.impl.bavet.common.AbstractNodeBuildHelper;
+import ai.timefold.solver.core.impl.bavet.uni.AbstractForEachUniNode;
+import ai.timefold.solver.core.impl.move.streams.dataset.common.DataNodeBuildHelper;
+
+public final class DatasetSessionFactory<Solution_> {
+
+    private final DefaultDataStreamFactory<Solution_> dataStreamFactory;
+
+    public DatasetSessionFactory(DefaultDataStreamFactory<Solution_> dataStreamFactory) {
+        this.dataStreamFactory = dataStreamFactory;
+    }
+
+    public DatasetSession<Solution_> buildSession() {
+        var activeDataStreamSet = new LinkedHashSet<AbstractDataStream<Solution_>>();
+        var datasets = dataStreamFactory.getDatasets();
+        for (var dataset : datasets) {
+            dataset.collectActiveDataStreams(activeDataStreamSet);
```

Comment:
I wonder if we should change the logic to return the list of active datasets. This way, we wouldn't need to pass activeDatasetStreamSet as a parameter.

Reply:
It is a sort of a visitor pattern. Personally, I don't mind, and making this change would introduce yet another completely unrelated change to this already large PR. 

It's been like this forever, I'm keeping it.